### PR TITLE
Bump kubernetes dependencies to v0.31.x.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Bump kubernetes dependencies to v0.31.x.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Related to https://github.com/kubernetes-sigs/jobset/pull/662#issuecomment-2323440449.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -51,8 +51,8 @@ var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	// Example: {Default: true, PreRelease: featuregate.Beta},
 }
 
-func SetFeatureGateDuringTest(tb testing.TB, f featuregate.Feature, value bool) func() {
-	return featuregatetesting.SetFeatureGateDuringTest(tb, utilfeature.DefaultFeatureGate, f, value)
+func SetFeatureGateDuringTest(tb testing.TB, f featuregate.Feature, value bool) {
+	featuregatetesting.SetFeatureGateDuringTest(tb, utilfeature.DefaultFeatureGate, f, value)
```
Comment (mbobrovskyi) [inline pkg/features/features.go:55]: I also can't find where this function is being used. Maybe we can just remove it.

Comment (kannon92) [reply pkg/features/features.go:55]: It was added so that when we use features we can use this.

Comment (kannon92) [reply pkg/features/features.go:55]: That being said, it’s not hard to add back. @ahg-g ?

Comment (googs1025) [reply pkg/features/features.go:55]: It was added in this PR https://github.com/kubernetes-sigs/jobset/pull/557/files. At that time, the feature gate feature was supported, but this feature has not been used in the current project. If it needs to be deleted, I think it is reasonable.

Comment (ahg-g) [reply pkg/features/features.go:55]: it is fine, we can keep it.

**Comment by ahg-g** [comment]
/lgtm
/approve 

Thanks a lot for fixing this!

File: Makefile [hunk 2]
```diff
@@ -124,7 +124,7 @@ ci-lint: golangci-lint
 	$(GOLANGCI_LINT) run --timeout 15m0s
 
 .PHONY: test
-test: manifests fmt vet envtest gotestsum test-python-sdk
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply Makefile:127]: Uhm... It seems it was accidental. I will undo this change.

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply Makefile:127]: Reverted on https://github.com/kubernetes-sigs/jobset/pull/673.

**Pull request merged**